src/Tasks.UnitTests/Delete_Tests.cs

```diff
@@ -34,5 +37,45 @@ public void AttributeForwarding()
             // Output ItemSpec should not be overwritten.
             Assert.Equal("MyFiles.nonexistent", t.DeletedFiles[0].ItemSpec);
         }
+
+        /// <summary>
+        /// Retry Delete
+        /// </summary>
+        [Fact]
+        public void DeleteWithRetries()
+        {
+            string source = FileUtilities.GetTemporaryFile();
+            try
+            {
+                using (StreamWriter sw = FileUtilities.OpenWrite(source, true))
+                {
+                    sw.Write("This is a source file.");
+                }
+
+                File.SetAttributes(source, FileAttributes.ReadOnly);
+
+                ITaskItem sourceItem = new TaskItem(source);
+                ITaskItem[] sourceFiles = { sourceItem };
+
+                var t = new Delete
+                {
+                    RetryDelayMilliseconds = 1,  // speed up tests!
+                    BuildEngine = new MockEngine(),
+                    Files = sourceFiles,
+                    Retries = 1,
+                };
+
+                // Should fail since file is readonly
```

Comment:
If you want to be really fancy, you could have something that waits a second after making the file readonly, then makes it not readonly, and verify that it initially fails but eventually succeeds. I don't think that's necessary here, though 🙂

Reply:
Thanks for pointing it out. I also want to do it ealier, but also think it's not necessary. But now I have updated the test case as what you said.